src/util/Constants.js

```diff
@@ -1269,13 +1269,56 @@ exports.InteractionResponseTypes = createEnum([
  * The type of a message component
  * * ACTION_ROW
  * * BUTTON
- * * SELECT_MENU
  * * TEXT_INPUT
+ * * STRING_SELECT
+ * * USER_SELECT
+ * * ROLE_SELECT
+ * * MENTIONABLE_SELECT
+ * * CHANNEL_SELECT
+ * * SELECT_MENU (deprecated)
  * @typedef {string} MessageComponentType
  * @see {@link https://discord.com/developers/docs/interactions/message-components#component-object-component-types}
  */
-exports.MessageComponentTypes = createEnum([null, 'ACTION_ROW', 'BUTTON', 'SELECT_MENU', 'TEXT_INPUT']);
-
+exports.MessageComponentTypes = {
+  ...createEnum([
+    null,
+    'ACTION_ROW',
+    'BUTTON',
+    'STRING_SELECT',
```

Comment:
This is a breaking change
```js
// before
MessageComponentTypes[3]; // 'SELECT_MENU'

// now
MessageComponentTypes[3]; // 'STRING_SELECT'
```

Reply:
yeah 🙄

Reply:
*good question*

Reply:
_not a question_

Reply:
some sollution?

Reply:
however we cant do anything

Reply:
You would need to swap `STRING_SELECT` and `SELECT_MENU`, although the deprecation will not longer be set.

If you swap, `MessageComponentTypes.SELECT_MENU` and `MessageComponentTypes.STRING_SELECT` will return `3`, just like it does now, but `MessageComponentTypes[3]` will return `SELECT_MENU`, just like it does before this patch, and that will resolve the semver-major aspect of this PR.